src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -183,7 +183,11 @@ public override void OnRemoveBehavior()
 
             if (updateUserStats)
             {
-                userUpdate.Rating = GetNewRating(crpgPeer);
+                var rating = GetNewRating(crpgPeer, out int ratedCharacterId);
+                if (userUpdate.CharacterId == ratedCharacterId)
+                {
+                    userUpdate.Rating = rating;
+                }

```

Comment:
You are always returning `crpgPeer.User!.Character.Id` from `GetNewRating` so this check is always true

Reply:
I don't think so. The original bug happens when people switch character. So my understanding was that the dictionary had the rating changes for the old character , but userUpdate.CharacterId Would be the new character. As we were fetching peerId before , peerId did not change when character changed , so the rewardServer happily applied the rating changes to the new character because for the rewardServer it was the same peerId

Reply:
You're right. I do not need neither the out variable nor the check